src/platform/web/dom/History.js

```diff
@@ -30,10 +30,10 @@ export class History extends BaseObservableValue {
         But for SSO, we need to handle <root>/?loginToken=<TOKEN>
         Handle that as a special case for now.
         */
-        if (document.location.search.includes("loginToken")) {
+        if (document?.location?.search.includes("loginToken")) {
             return document.location.search;
         }
-        return document.location.hash;
+        return document?.location?.hash;
```

Comment:
This is to allow this function to run when server-side rendered in `linkedom`. It's ok for this function to return `undefined`. See [`shared/lib/archive-history.js`](https://github.com/matrix-org/matrix-public-archive/pull/12/files#diff-511a3e74302f787f8e9c5b9a97c92ecee3ef3706a917b3134a2c5b5c5033b561R1) for how this is integrated.

If this isn't desired, I can always wrap `super.get()` in my custom `History` instance and default that way.

Reply:
I think it'd be better if you add your custom logic in your custom instance. It's going to be hard to not have regressions for some arcane DOM impl we never test for.